core/src/main/scala/filodb.core/columnstore/InMemoryColumnStore.scala

```diff
@@ -37,7 +37,11 @@ extends CachedMergingColumnStore with StrictLogging {
 
   def initializeProjection(projection: Projection): Future[Response] = Future.successful(Success)
 
-  def clearProjectionDataInner(projection: Projection): Future[Response] = ???
+  def clearProjectionDataInner(projection: Projection): Future[Response] = Future {
+    chunkDb.keys.filter { case (ds, _, _) => ds == projection.dataset }.foreach(chunkDb.remove)
+    rowMaps.keys.filter { case (ds, _, _) => ds == projection.dataset }.foreach(rowMaps.remove)
+    Success
+  }
```

Comment:
Maybe this and iterate once through each key collection using 'collect' versus twice using 'filter' + 'foreach'
 
     def clearProjectionDataInner(projection: Projection): Future[Response] = Future {
         chunkDb.keys.collect { case key if key._1 == projection.dataset => chunkDb remove key }
         rowMaps.keys.collect { case key if key._1 == projection.dataset => rowMaps remove key }
        Success
     }